src/udps.cpp

```diff
@@ -246,21 +223,60 @@ size_t Udp::deliver(size_t maxcalls)
         }
     }
 
-    // something was processed
+    // done
+    return result;
+}
+
+/**
+ *  Invoke callback handlers for buffered raw responses
+ *  @param   watcher   to keep track if the parent object remains valid
+ *  @param   maxcalls  the max number of callback handlers to invoke
+ *  @return  number of callback handlers invoked
+ */
+size_t Udps::deliver(size_t maxcalls)
+{
+    // result variable
+    size_t result = 0;
+
+    // we are going to make a call to userspace, so we keep monitoring if `this` is not destructed
+    Watcher watcher(this);
+
+    // deliver from all sockets
+    for (auto &socket : _sockets) result += socket.deliver(&watcher, maxcalls);
+
```

Comment:
Between each call to socker.deliver() the watcher must be checked because user-space might have destructed `this`